server/app/services/image/image.py

```diff
@@ -1,15 +1,54 @@
-import cv2
+from dotenv import load_dotenv
+import json
+import os
+from pathlib import Path
+from PIL import Image, ImageDraw, ImageFont
+from web3 import Web3
 
 
-def generate_image(token_id: str):
-    deposited_token_address = token_id[:42]
-    deposited_value = token_id[42:]
+load_dotenv(Path("../../../../.env"))
+INFURA_ENDPOINT = os.getenv('INFURA_GOERLI_ENDPOINT')
 
-    # from token address get token name
+def get_token_info(token_address: str):
+    with open('erc20.json') as r:
+        abi = json.loads(r.read())
+    w3 = Web3(Web3.HTTPProvider(INFURA_ENDPOINT))
+    token = w3.eth.contract(w3.toChecksumAddress(token_address), abi=abi)
+    try:
+        name = token.functions.name().call()
+    except:
+        name = 'Unknown'
+    try:
+        symbol = token.functions.symbol().call()
+    except:
+        symbol = '?'
+    return name, symbol
 
-    # load base ticket image
+def generate_image(token_id: int):
+    token_address = hex(int(token_id >> 96 & ((1 << 160) - 1)) - (token_id >> 96 & (1 << 160)))
+    if token_address != '0x0':
+        token_name, token_symbol = get_token_info(token_address)
+        token_amount = (token_id & ((1 << 96) - 1)) - (token_id & (1 << 96))
+    else:
+        token_name = "Ethereum"
```

Comment:
A bit nit-picky, but it should be "ETHER". Ether is the token name, while Ethereum is the network

Reply:
Good call 👍 whoops!